examples/js/loaders/STLLoader.js

```diff
@@ -222,7 +222,7 @@ THREE.STLLoader.prototype = {
 
 			}
 
-			patternVertex = /vertex[\s]+([\-+]?[0-9]+\.?[0-9]*([eE][\-+]?[0-9]+)?)+[\s]+([\-+]?[0-9]*\.?[0-9]+([eE][\-+]?[0-9]+)?)+[\s]+([\-+]?[0-9]*\.?[0-9]+([eE][\-+]?[0-9]+)?)+/g;
+			patternVertex = /vertex[\s]+([\-+]?[0-9]+\.?[0-9]*([eE][\-+]?[0-9]+)?)+[\s]+([\-+]?[0-9]*\.?[0-9]*([eE][\-+]?[0-9]+)?)+[\s]+([\-+]?[0-9]*\.?[0-9]*([eE][\-+]?[0-9]+)?)+/g;
```

Comment:
1. You are right, but if we would just change that to "*"  then it would be also possible to match only a dot (".") as valid "number" which is of course wrong. But yeah .3 should be also allowed but that means that we would have to lookahead

2.  you are right

3. Also correct 

Here another regex which includes your points:
vertex([\s]+([\-+]?[0-9]*\.?[0-9]*([eE][\-+]?[0-9]+)?)){3}

The problem with this is that "vertex . 0.1 -4." would also be valid because of the reason mentioned in point 1. But haven't worked with lookahead in regex yet so if you a into it, please feel free to contribute ;)